compiler/quilt/test/test_import.py

```diff
@@ -351,6 +355,72 @@ def test_set_non_node_attr(self):
         with self.assertRaises(AttributeError):
             package4.newdf = df
 
+    def test_load_update(self):
+        # also tests dynamic import
+        mydir = os.path.dirname(__file__)
+        build_path = os.path.join(mydir, './build.yml')
+        command.build('foo/package5', build_path)
+        from quilt.data.foo import package5
+
+        # make a copy, to prove we can
+        newpkgname = 'foo/copied_package'
+        quilt.build(newpkgname, package5)
+
+        newfilename = 'myfile'+str(int(time.time()))
+        with open(newfilename, 'w') as fh:
+            fh.write('hello world1')
+
+        module = quilt.load(newpkgname)
+        module._set([newfilename], newfilename)
+        quilt.build(newpkgname, module)
+
+        # current spec requires that build() *not* update the in-memory module tree.
+        newpath1 = getattr(module, newfilename)()
+        assert newpath1 == newfilename
+
+        # current spec requires that load() reload from disk, i.e. gets a reference
+        # to the local object store
+        # this is important because of potential changes to myfile
+        reloaded_module = quilt.load(newpkgname)
+        assert reloaded_module is not module
+        newpath2 = getattr(reloaded_module, newfilename)()
+        assert 'myfile' not in newpath2
+
+    def test_multiple_updates(self):
+        mydir = os.path.dirname(__file__)
+        build_path = os.path.join(mydir, './build.yml')
+        command.build('foo/package', build_path)
+        from quilt.data.foo import package
```

Comment:
does quilt.import() avoid caching problems?

Reply:
On cursory inspection, `quilt.load` appears to avoid those issues, yes.

However, this also means that calling `quilt.load()` twice means you get different (but identical) objects, and data isn't persisted unless you save the object using `quilt.build()`.